solutions/swb-app/src/userRoutes.ts

```diff
@@ -7,39 +7,153 @@
 import {
   CreateRoleSchema,
   CreateUserSchema,
+  Status,
   UpdateRoleSchema,
-  UserManagementService
+  UserManagementService,
+  isUserNotFoundError,
+  isRoleNotFoundError,
+  isInvalidParameterError,
+  isUserAlreadyExistsError,
+  UpdateUserRequest,
+  UpdateUserRequestParser
 } from '@aws/workbench-core-authentication';
+import { validateAndParse } from '@aws/workbench-core-base';
 import Boom from '@hapi/boom';
 import { Request, Response, Router } from 'express';
 import { validate } from 'jsonschema';
 import _ from 'lodash';
 import { wrapAsync } from './errorHandlers';
 import { processValidatorResult } from './validatorHelper';
 
-export function setUpUserRoutes(router: Router, user: UserManagementService): void {
+export function setUpUserRoutes(router: Router, userService: UserManagementService): void {
   router.post(
     '/users',
     wrapAsync(async (req: Request, res: Response) => {
-      processValidatorResult(validate(req.body, CreateUserSchema));
-      const response = await user.createUser(req.body);
-      res.status(201).send(response);
+      try {
+        processValidatorResult(validate(req.body, CreateUserSchema));
+        const response = await userService.createUser(req.body);
+        res.status(201).send(response);
+      } catch (err) {
+        if (isInvalidParameterError(err)) {
+          throw Boom.badRequest(`Invalid parameter: ${err.message}`);
+        }
+
+        if (isUserAlreadyExistsError(err)) {
+          throw Boom.conflict(`User with this email already exist.`);
+        }
+
+        if (Boom.isBoom(err)) {
+          throw err;
+        }
+
+        throw Boom.badImplementation(`Could not find create user`);
```

Comment:
maybe something like "Could not create user" would be better. The current message isnt clear to me.

Reply:
Copy-paste issue  , neither clear to me  😄